website/src/docs/tus.md

```diff
@@ -87,9 +87,29 @@ When uploading a chunk fails, automatically try again after the millisecond inte
 
 Set to `null` to disable automatic retries, and fail instantly if any chunk fails to upload.
 
-### `retryStatusCodes: [429]`
+### `onShouldRetry: (err, next) => next(err)`
 
-Tus can retry with [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) when receiving 429 (Too Many Requests), or other configured status codes. This can be useful to retry when you hit an API rate limit or when your authentication token is expired.
+When an upload fails `onShouldRetry` is called with the error and the default retry logic as the second argument. The default retry logic is an [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) algorithm triggered on HTTP 429 (Too Many Requests) errors. Meaning if your server (or proxy) returns HTTP 429 because it’s being overloaded, @uppy/tus will find the ideal sweet spot to keep uploading without overloading.
+
+If you want to extend this functionality, for instance to retry on unauthorized requests (to retrieve a new authentication token):
+
+```js
+import Uppy from '@uppy/core'
+import Tus from '@uppy/tus'
+
+const uppy = new Uppy().use(Tus, { endpoint: '', onShouldRetry })
+
+async function onShouldRetry (err) {
```

Comment:
> We wrap the function in @uppy/tus so if that one waits for a promise we should be okay?

No I don't think we would be OK, tus would see a `Promise` object as a return value and interpret it as a truthy value, so it would always retry even if you `return Promise.resolve(false)` or even `return Promise.reject()`.